larq_compute_engine/mlir/tests/legalize-lce.mlir

```diff
@@ -0,0 +1,28 @@
+// RUN: lce-tf-opt %s -tfl-legalize-lce | FileCheck %s --dump-input-on-failure
+
+// CHECK-LABEL: @legalize_bconv2d
+func @legalize_bconv2d(%arg0: tensor<256x32x32x3xf32>, %arg1: tensor<16x3x3x3xf32>, %arg2: tensor<16xf32>, %arg3: tensor<16xf32>, %arg4: none) -> tensor<256x30x30x16xf32> {
+  %0 = "lq.Bconv2d"(%arg0, %arg1, %arg2, %arg3, %arg4) {channels_in = 3 : i32, dilation_height_factor = 1 : i32, dilation_width_factor = 1 : i32, fused_activation_function = "NONE", padding = "VALID", stride_height = 1 : i32, stride_width = 1 : i32} : (tensor<256x32x32x3xf32>, tensor<16x3x3x3xf32>, tensor<16xf32>, tensor<16xf32>, none) -> tensor<256x30x30x16xf32>
+  return %0 : tensor<256x30x30x16xf32>
+
+  // CHECK: %0 = "tfl.custom"(%arg0, %arg1, %arg2, %arg3, %arg4) {custom_code = "LceBconv2d", custom_option = opaque<"lq", "0x6368616E6E656C735F696E0064696C6174696F6E5F6865696768745F666163746F720064696C6174696F6E5F77696474685F666163746F720066757365645F61637469766174696F6E5F66756E6374696F6E00044E4F4E45007061645F76616C7565730070616464696E67000556414C4944007374726964655F686569676874007374726964655F776964746800088F846E593A30221508010803010149003201010404041404140404102401"> : tensor<173xi8>} : (tensor<256x32x32x3xf32>, tensor<16x3x3x3xf32>, tensor<16xf32>, tensor<16xf32>, none) -> tensor<256x30x30x16xf32>
```

Comment:
Wow that `custom_option` thing is very ugly. Is there any point matching against that?

Reply:
> Wow that custom_option thing is very ugly

Yes, this is the raw data of the flexbuffer which will be stored in the `.tflite` file.

> Is there any point matching against that?

I added this mainly as a sanity check to make sure that the test fails when we change attributes. I actually manually checked that this is the correct representation of the attributes.

Reply:
Yeah makes sense, my only concern would be that when the op attributes change I'm not convinced that people will manually check correctness of the new values vs. blindly copying in what MLIR reports the output to be. That said, I guess this is really what the End2End tests are designed for, so it shouldn't be too concerning.